### DISCUSSION TITLE
Extension Issue

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11590] [Q&A]
Hi,
I'm trying to work with this extension in Theia. However, it isn't wokring as expected.

In VSCode, 
After installing this extension, opening a text file with Hex Editor seems to be working fine.
https://marketplace.visualstudio.com/items?itemName=ms-vscode.hexeditor

However, the same in Theia seems to be an issue. 
https://open-vsx.org/extension/ms-vscode/hexeditor

Kindly confirm if this is an issue with the extension or with Theia.

Thanks

**Comment by vince-fugnitto** [comment]
@maheshrajrp thank you for the discussion, do you mind providing additional information?
- what did not work well for you?
- did you notice any errors, logs in both the backend or frontend?

I installed the extension, and triggered `HexEditor: Open File` and the extension correctly installed and started (tested on master):

![Screen Shot 2022-08-22 at 8 02 58 AM](https://user-images.githubusercontent.com/40359487/185916733-38347b01-165f-4c7e-8dbb-157b6dc87fc9.png)

**Reply by maheshrajrp** [reply]
Hi @vince-fugnitto , 
Thanks for your reply. In that case, I believe that to be browser based issue.

Steps to Reproduce:
1. Run Theia in GitPod
2. Right Click any Text File and Open with Hex Editor.
3. You might see nothing is displayed in the newly opened editor;

Further Info:
OS: Linux CentOS 8
Browser: Chrome
Theia: v1.28.0

The below image shows the text file that is opened with the HexEditor along with its HexEditor.
![image](https://user-images.githubusercontent.com/80001447/186049194-b9ed2e45-b53e-4361-b56a-21699ddf54a4.png)

**Reply by msujew** [reply]
Hey @maheshrajrp,

can you try running Theia locally instead? Theia uses special fake URLs to load webviews. However, the domain names used are not exposed by Gitpod, which is why every request to a webview will return a `404` not found error.

**Reply by maheshrajrp** [reply]
Hi @msujew,
Thanks for responding. Sorry for delayed reply. I'm facing the some issue in local too. I wanted to check more on that and that is why I tried with GitPod. 

Steps to Replicate:
1. Select a text file & Open With ~> Hex Editor in File Explorer Widget.

OS: CentOS 8
Broswer: Chrome
Theia: v1.28

![image](https://user-images.githubusercontent.com/80001447/186447600-adafee3b-7c3c-4728-ba54-7fbb80f3a7ad.png)

**Reply by vince-fugnitto** [reply]
@maheshrajrp it seems to work fine for me, I am able to open the hex editor using `open with...` in the browser locally:

https://user-images.githubusercontent.com/40359487/186658550-d704d0be-f3ba-4562-bb50-f7bd5d5f076a.mp4
